pkg/types/installconfig.go

```diff
@@ -105,6 +105,9 @@ type InstallConfig struct {
 	// When no strategy is specified, the strategy is `External`.
 	// +optional
 	Publish PublishingStrategy `json:"publish,omitempty"`
+
+	// FIPS configures https://en.wikipedia.org/wiki/Federal_Information_Processing_Standards
+	FIPS *bool `json:"fips,omitempty"`
```

Comment:
Why boolean ptr ?

Also why is this not a installer-machine-pool configuration like hyperthreading?

Reply:
I couldn't think of a use case for a "partially FIPS" cluster, whereas it's a clear use case to e.g. just disable hyperthreading on workers, or a specific pool of workers.

(That said I think what we really want is something more like https://lwn.net/Articles/799454/ where "hyperthreading" is determined automatically - we have all of the info for it.  But anyways)